### DISCUSSION TITLE
ODB::Several Chips ?

### DISCUSSION THREAD

**Discussion by ApeachM** [The-OpenROAD-Project/OpenROAD#3769] [Q&A] [answered]
I can see a function of `dbDatabase`, which is 

`dbSet<dbChip> dbDatabase::getChips()`.

However, in `dbChip* dbChip::create(dbDatabase* db_)` function,
```cpp
///
/// Create a new chip.
/// Returns nullptr if a chip already exists.
/// Returns nullptr if there is no database technology.
///
dbChip* dbChip::create(dbDatabase* db_)
{
  _dbDatabase* db = (_dbDatabase*) db_;

  if (db->_chip != 0)
    return nullptr;

  _dbChip* chip = db->_chip_tbl->create();
  db->_chip = chip->getOID();
  return (dbChip*) chip;
}
```
it prevents making several Chip object.
Is `getChips()` are deprecated one and there's no way to make several chip in current odb version?
Or is there any way to use `dbSet<dbChip> dbDatabase::getChips()` function and handle several Chip object?

I appreciate for spending your precious time reading my question. Thank you.

**Comment by maliberty** [comment, marked as answer]
The dbChip is a pretty useless class.  You can only create one but you can create multiple dbBlock in one dbChip.  I expect to model all the die/tier/stack as dbBlock and mostly ignore dbChip.

**Reply by maliberty** [reply]
I'd like to remove dbChip at some point but that will require touching a lot of code as it will be a breaking API change.

**Reply by ApeachM** [reply]
Thank you for the good explanation. I was always wondering about the role of `dbChip` class. Thank you!
